### PR TITLE
Add new ModuleBuild and ModulePackage models.

### LINKED ISSUE #1324: AddModuleModels
In order to support modules, we need to add new database models. Before we can do this, we need to modify Bodhi's existing models to be multi-type friendly. Thus, the following modules must be converted before this issue can be worked on:

- [x] Package - #1392
- [x] Build - #1393
- [x] Update - #1394
- [x] BuildrootOverride - #1395

Each of these models will need to be reworked to handle multiple types by splitting them into superclass/subclass (with the subclass being the RPM-specific bits of the current model), and database migrations will be needed. When we are done with this phase, Bodhi should still work like it does today with RPMs.

Once each of these models are able to support multiple types, we will next be able to add more models to satisfy this ticket:

- [x] Module
- [x] ModuleBuild

### LINKED ISSUE #1459: We need to validate that RpmBuilds are only associated with RpmPackages and vice-versa
This is also true for ModuleBuilds and ModulePackages, though we can solve that one in #1324.

### PR DESCRIPTION
**Pull request by bowlofeggs**
fixes #1324
fixes #1459

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

File: bodhi/server/models.py [hunk 1]
```diff
@@ -729,6 +765,20 @@ def unpush(self, koji):
...
+    """
+    Represents a Module build.
+
+    Note that this model uses single-table inheritance with its Build superclass.
+
+    Attributes:
+        nvr (unicode): A unique Koji identifier for the module build.
+    """
+    __mapper_args__ = {
+        'polymorphic_identity': ContentType.module,
```
Comment (ralphbean) [inline bodhi/server/models.py:778]: Fancy!

**Review by ralphbean — APPROVED** [review]
You, sir, are a gentleman and a scholar.

File: bodhi/tests/server/test_models.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ class TestQueryProperty(BaseTestCase):
 
     def test_session(self):
         """Assert the session the query property uses is from the scoped session."""
-        query = model.Package.query
+        query = model.RpmPackage.query
```
Comment (jeremycline) [inline bodhi/tests/server/test_models.py]: Out of curiosity, why did this need to be changed?

**Review by jeremycline — APPROVED** [review]
Very nice, :shipit:

**Comment by ralphbean** [comment]
OK to merge?

File: bodhi/tests/server/test_models.py (hunk 2 above)
Comment (bowlofeggs) [reply bodhi/tests/server/test_models.py]: @jeremycline I'm not sure I can recall why it needed to be changed. Presumably some test was failing at some point, but when I change it back the tests all pass so I'll set it back. Thanks for pointing that out!

**Comment by bowlofeggs** [comment]
@ralphbean Yeah I'm working on merging this today, then we can move on to looking at the API.

**Pull request merged**